react/components/Upload/Upload.tsx

```diff
@@ -0,0 +1,185 @@
+import * as React from 'react'
+import { useMutation } from 'react-apollo'
+import { useDropzone } from 'react-dropzone'
+import { defineMessages, useIntl } from 'react-intl'
+import {
+  Button,
+  IconClose,
+  IconImage,
+  Input,
+  Spinner,
+  IconCaretDown,
+} from 'vtex.styleguide'
+
+import StyleButton from './StyleButton'
+import UploadFileMutation from '../../graphql/uploadFile.graphql'
+import { FormRawInputProps, InputTypes } from '../../typings/InputProps'
+import { HiddenInput } from '../Input'
+
+interface MutationData {
+  uploadFile: { fileUrl: string }
+}
+
+const MAX_SIZE = 4 * 1024 * 1024
+
+const messages = defineMessages({
+  addBtn: {
+    defaultMessage: 'Add',
+    id: 'admin/pages.admin.rich-text-editor.add-button',
+  },
+  addTitle: {
+    defaultMessage: 'Insert document',
+    id: 'admin/pages.admin.rich-text-editor.add-document.title',
+  },
+  addLabel: {
+    defaultMessage: 'Document URL',
+    id: 'admin/pages.admin.rich-text-editor.add-document.label',
+  },
+  addPlaceholder: {
+    defaultMessage: 'URL',
+    id: 'admin/pages.admin.rich-text-editor.add-document.placeholder',
+  },
+  fileSizeError: {
+    defaultMessage:
+      'File exceeds the size limit of 4MB. Please choose a smaller one.',
+    id: 'admin/pages.editor.document-uploader.error.file-size',
+  },
+  genericError: {
+    defaultMessage: 'Something went wrong. Please try again.',
+    id: 'admin/pages.editor.document-uploader.error.generic',
+  },
+  uploadBtn: {
+    defaultMessage: 'Upload document',
+    id: 'admin/pages.admin.rich-text-editor.upload-button',
+  },
+  uploadLabel: {
+    defaultMessage: 'Choose an document to upload',
+    id: 'admin/pages.admin.rich-text-editor.upload-document-label',
+  },
+})
```

Comment:
All these messages are necessary? And please change the ids, they all are related to admin rich-text-editor

Reply:
Yes, they are. I updated the ids to store-form pattern